File: src/main/java/delta/module/modules/HUD.java
package delta.module.modules;

import delta.DeltaCore;
import delta.module.Category;
import delta.module.Module;
import net.minecraft.client.renderer.GlStateManager;
import net.minecraftforge.client.event.RenderGameOverlayEvent;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.awt.*;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.stream.Collectors;

public class HUD extends Module {
    public HUD() {
        super("HUD", "Heads up display", Category.HUD);
    }

    public void onEnable() {
        MinecraftForge.EVENT_BUS.register(this);
    }

    public void onDisable() {
        MinecraftForge.EVENT_BUS.unregister(this);
    }

    @SubscribeEvent
    public void onRender2D(RenderGameOverlayEvent.Text event) {
        int y = 520 - ((mc.fontRenderer.FONT_HEIGHT + 4) * getSorted().size());
        for (Module module : getSorted()) {
            mc.fontRenderer.drawString(module.getFullString(), 910 - mc.fontRenderer.getStringWidth(module.getFullString()), y, new Color(255, 255, 255).hashCode());
            y = y + mc.fontRenderer.FONT_HEIGHT + 4;
        }
//                GlStateManager.scale(2, 2, 2);
//        mc.fontRenderer.drawString("Delta", 10, 10, new Color(255,255,255).hashCode());
//        GlStateManager.scale(2.5, 2.5, 2.5);
//        mc.fontRenderer.drawString("Delta", 7, 7, new Color(0,0,0).hashCode());
//        GlStateManager.scale(0.5, 0.5, 1);
        mc.fontRenderer.drawString("[ Delta - welcome to the botnet " + mc.session.getUsername() + " ]", 10, 10, new Color(255, 255, 255).hashCode());
    }

    public ArrayList<Module> getSorted() {
        return (ArrayList<Module>) DeltaCore.moduleManager.getEnabledModules().stream().sorted(Comparator.comparing(m -> mc.fontRenderer.getStringWidth(m.getFullString()))).collect(Collectors.toList());
    }
}

File: src/main/java/delta/module/modules/HUD.kt
package delta.module.modules

import delta.DeltaCore
import delta.module.Category
import delta.module.Module
import delta.setting.Setting
import delta.util.RenderUtils
import net.minecraftforge.client.event.RenderGameOverlayEvent
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import java.awt.Color

class HUD: Module("HUD", "wtf hud!", Category.RENDER) {
    var x: Setting = setting("X", 1000.0, 0.0, 1000.0, true)
    var lowercase: Setting = setting("Lowercase", true)
    var line = setting("Line", true)
    var r = setting("Red", 1.0, 1.0, 255.0, true)
    var g = setting("Green", 1.0, 1.0, 255.0, true)
    var b = setting("Blue", 1.0, 1.0, 255.0, true)
    var alpha = setting("Alpha", 1.0, 1.0, 255.0, true)
    override fun onEnable() {
        MinecraftForge.EVENT_BUS.register(this)
    }

    override fun onDisable() {
        MinecraftForge.EVENT_BUS.unregister(this)
    }

    @SubscribeEvent
    fun onRender2D(event: RenderGameOverlayEvent.Text) {
        val startColor = Color(r.dVal.toInt(), g.dVal.toInt(), b.dVal.toInt())
        val endColor = Color(r.dVal.toInt(), g.dVal.toInt(), b.dVal.toInt())
        var k = 0
        for (module in DeltaCore.moduleManager.sorted()) {
            for (i in 0..k) {
                endColor.darker()
            }
            RenderUtils.drawRect(
                (x.dVal - DeltaCore.fontRenderer.getStringWidth(module.fullString) - 10).toInt(),
                (10 + ((DeltaCore.fontRenderer.height + 2 ) * k)),
                DeltaCore.fontRenderer.getStringWidth(module.fullString) + 11,
                DeltaCore.fontRenderer.height + 2,
                Color(0, 0, 0, alpha.dVal.toInt())
            )
            DeltaCore.fontRenderer.drawString(
                lowcase(module.name),
                (x.dVal - DeltaCore.fontRenderer.getStringWidth(module.fullString)).toFloat(),
                (10 + ((DeltaCore.fontRenderer.height + 2 ) * k)).toFloat(),
                endColor.hashCode()
            )
            module.hudString?.let {
                DeltaCore.fontRenderer.drawString(
                    lowcase(" " + module.hudString),
                    (x.dVal - DeltaCore.fontRenderer.getStringWidth(module.hudString) - 2).toFloat(),
                    (10 + ((DeltaCore.fontRenderer.height + 2 ) * k)).toFloat(),
                    Color(255, 255 , 255).hashCode()
                )
            }
            k++
        }
    }

    fun lowcase(string: String): String {
        if (lowercase.bVal) {
            return string.toLowerCase()
        }
        return string
    }
}